src/main/kotlin/com/terraformation/backend/nursery/db/BatchStore.kt

```diff
@@ -270,19 +274,25 @@ class BatchStore(
           .execute()
 
       // Withdrawals that are only from this batch should be deleted.
-      dslContext
-          .deleteFrom(WITHDRAWALS)
-          .where(
-              WITHDRAWALS.ID.`in`(
-                  DSL.select(BATCH_WITHDRAWALS.WITHDRAWAL_ID)
+      val withdrawalIds =
+          dslContext
+              .select(WITHDRAWALS.ID)
+              .from(WITHDRAWALS)
+              .join(BATCH_WITHDRAWALS)
+              .on(WITHDRAWALS.ID.eq(BATCH_WITHDRAWALS.WITHDRAWAL_ID))
+              .where(BATCH_WITHDRAWALS.BATCH_ID.eq(batchId))
+              .andNotExists(
+                  DSL.selectOne()
                       .from(BATCH_WITHDRAWALS)
-                      .where(BATCH_WITHDRAWALS.BATCH_ID.eq(batchId))))
-          .andNotExists(
-              DSL.selectOne()
-                  .from(BATCH_WITHDRAWALS)
-                  .where(WITHDRAWALS.ID.eq(BATCH_WITHDRAWALS.WITHDRAWAL_ID))
-                  .and(BATCH_WITHDRAWALS.BATCH_ID.ne(batchId)))
-          .execute()
+                      .where(WITHDRAWALS.ID.eq(BATCH_WITHDRAWALS.WITHDRAWAL_ID))
+                      .and(BATCH_WITHDRAWALS.BATCH_ID.ne(batchId)))
+              .fetch(WITHDRAWALS.ID.asNonNullable())
+
+      if (withdrawalIds.isNotEmpty()) {
```

Comment:
we could skip this check since foreach over empty is no-op-ish anyways..?

Reply:
I needed it for the `DELETE` statement (an empty `IN` clause is invalid SQL) and it seemed slightly cleaner to wrap the `forEach` at the same time.

Reply:
ok, got it,, makes total sense 👍